examples/dart/test/open_flexible_sync_realm_test.dart

```diff
@@ -35,6 +34,61 @@ void main() {
       expect(realm.isClosed, true);
       expect(app.currentUser, null);
     });
+    test('Async Open Flexible Sync Realm', () async {
+      Credentials credentials = Credentials.anonymous();
+      User currentUser = await app.logIn(credentials);
+      // :snippet-start: async-open
+      Configuration config =
+          Configuration.flexibleSync(currentUser, [Tricycle.schema]);
+      Realm fullySyncedRealm = await Realm.open(config);
```

Comment:
I don't know what the common practice is for the docs, but I personally prefer to omit the types as it makes the code easier to read - e.g. here we have "Configuration config equals Configuration dot ...", "Realm realm equals await realm dot..." - that's a lot of repetition. I find the following a little lighter:

```suggestion
      final config = Configuration.flexibleSync(currentUser, [Tricycle.schema]);
      final realm = await Realm.open(config);
```

I also suggest renaming `fullySyncedRealm` as those snippets should be as copy-pastable as possible - I know we want to be explicit about what's going on there, but I'd suggest using the text around the snippet or possibly adding a comment rather than naming variables descriptive names.

Reply:
> I don't know what the common practice is for the docs, but I personally prefer to omit the types as it makes the code easier to read

lubo commented the same a few weeks ago. for the sake of consistency, i'm going to keep as is. but made ticket to update this everywhere across the flutter docs https://jira.mongodb.org/browse/DOCSP-26111

Reply:
> I also suggest renaming fullySyncedRealm as those snippets should be as copy-pastable as possible

and for this, sure can change